src/Model/User.php

```diff
@@ -1734,39 +1738,8 @@ public static function sendRegisterOpenEmail(L10n $l10n, array $user, string $si
 			$user['username'],
 			$sitename
 		));
-		$body = Strings::deindent($l10n->t(
-			'
-			The login details are as follows:
 
-			Site Location:	%3$s
-			Login Name:		%1$s
-			Password:		%5$s
-
-			You may change your password from your account "Settings" page after logging
-			in.
-
-			Please take a few moments to review the other account settings on that page.
-
-			You may also wish to add some basic information to your default profile
-			' . "\x28" . 'on the "Profiles" page' . "\x29" . ' so that other people can easily find you.
-
-			We recommend adding a profile photo, adding some profile "keywords" ' . "\x28" . 'very useful
-			in making new friends' . "\x29" . ' - and perhaps what country you live in; if you do not wish
-			to be more specific than that.
-
-			We fully respect your right to privacy, and none of these items are necessary.
-			If you are new and do not know anybody here, they may help
-			you to make some new and interesting friends.
-
-			If you ever want to delete your account, you can do so at %3$s/settings/removeme
-
-			Thank you and welcome to %2$s.',
-			$user['nickname'],
-			$sitename,
-			$siteurl,
-			$user['username'],
-			$password
-		));
+		$body = sprintf(self::welcome_text(), DI::baseUrl(), $user['nickname'], $password, DI::config()->get('config', 'sitename'));
```

Comment:
Move the `sprintf()` call into the `welcome_text()` method and provide `$user['nickname']` and `$password` as arguments.

```suggestion
		$body = self::welcome_text($user['nickname'], $password);
```